appeals/web/src/server/appeals/appeal-details/service-user/service-user.controller.js

```diff
@@ -0,0 +1,92 @@
+import { getAppealDetailsFromId } from '#appeals/appeal-details/appeal-details.service.js';
+import logger from '#lib/logger.js';
+import { addNotificationBannerToSession } from '#lib/session-utilities.js';
+import { changeServiceUserPage } from './service-user.mapper.js';
+import { updateServiceUser } from './service-user.service.js';
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const getChangeServiceUser = async (request, response) => {
+	return renderChangeServiceUser(request, response);
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderChangeServiceUser = async (request, response) => {
+	const {
+		errors,
+		params: { appealId, userType }
+	} = request;
+
+	const appealDetails = await getAppealDetailsFromId(request.apiClient, appealId);
+	if (userType in appealDetails) {
+		const mappedPageContents = changeServiceUserPage(
+			appealDetails,
+			request.session.updatedServiceUser,
+			userType,
+			errors
+		);
+
+		return response.render('patterns/change-page.pattern.njk', {
+			pageContent: mappedPageContents,
+			errors
+		});
+	}
+
+	return response.render('app/500.njk');
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const postChangeServiceUser = async (request, response) => {
+	request.session.updatedServiceUser = {
+		firstName: request.body['firstName'],
+		lastName: request.body['lastName'],
+		email: request.body['emailAddress']
+	};
+
+	const {
+		params: { appealId, userType }
+	} = request;
+
+	const appealDetails = await getAppealDetailsFromId(request.apiClient, appealId);
+
+	if (request.errors) {
+		return renderChangeServiceUser(request, response);
+	}
+
+	try {
+		// @ts-ignore
+		const serviceUserId = appealDetails[userType]?.id;
```

Comment:
What's the TS error being ignored here? It's a pain, but if it's possible to work around this with a JSDoc type or a type guard that would be preferable to the ts-ignore

Reply:
Because the userType is of type string but it is validated in the get to be a valid key of appealDetails. Problem is that using that check doesn't fix the TS error and I want to avoid listing the valid userTypes as more are coming (could be something to add once we have all the valid user types and data models)